### PR TITLE
Watch bookmarks documentation

### PR DESCRIPTION
**Pull request by wojtek-t**
/milestone 1.15

/assign @liggitt @lavalamp

### CODE REVIEW

File: content/en/docs/reference/using-api/api-concepts.md [hunk 1]
```diff
@@ -87,6 +87,7 @@ For example:
         ...
 
 A given Kubernetes server will only preserve a historical list of changes for a limited time. Clusters using etcd3 preserve changes in the last 5 minutes by default.  When the requested watch operations fail because the historical version of that resource is not available, clients must handle the case by recognizing the status code `410 Gone`, clearing their local cache, performing a list operation, and starting the watch from the `resourceVersion` returned by that new list operation. Most client libraries offer some form of standard tool for this logic. (In Go this is called a `Reflector` and is located in the `k8s.io/client-go/cache` package.)
+To mitigate the impact of short history window, we introduced a concept of `bookmark` watch event. It is a special kind of event to pass an information that all changes up to a given `resourceVersion` client is requesting has already been send. Those can be requested by `allowWatchBookmarks` option in watch requests, but clients shouldn't assume bookmarks are returned at any specific interval, nor may they assume the server will send any `bookmark` event. As of 1.15 release, it is an Alpha feature.
```
Comment (liggitt) [inline content/en/docs/reference/using-api/api-concepts.md]: Include the exact query parameter syntax (query param and value). Also include an example bookmark watch event so the type and content can be seen in docs.

**Comment by wojtek-t** [comment]
@liggitt - thanks, for comments, PTAL

File: content/en/docs/reference/using-api/api-concepts.md [hunk 2]
```diff
@@ -87,6 +87,24 @@ For example:
         ...
 
 A given Kubernetes server will only preserve a historical list of changes for a limited time. Clusters using etcd3 preserve changes in the last 5 minutes by default.  When the requested watch operations fail because the historical version of that resource is not available, clients must handle the case by recognizing the status code `410 Gone`, clearing their local cache, performing a list operation, and starting the watch from the `resourceVersion` returned by that new list operation. Most client libraries offer some form of standard tool for this logic. (In Go this is called a `Reflector` and is located in the `k8s.io/client-go/cache` package.)
+To mitigate the impact of short history window, we introduced a concept of `bookmark` watch event. It is a special kind of event to pass an information that all changes up to a given `resourceVersion` client is requesting has already been send. Object returned in that event is of the type requested by the request, but only `resourceVersion` field is set, e.g.:
+
```
Comment (liggitt) [inline content/en/docs/reference/using-api/api-concepts.md]: Missing the allow watch bookmarks param, right?

File: content/en/docs/reference/using-api/api-concepts.md [hunk 3]
```diff
@@ -87,6 +87,24 @@ For example:
...
+        Transfer-Encoding: chunked
+        Content-Type: application/json
+        {
+          "type": "ADDED",
+          "object": {"kind": "Pod", "apiVersion": "v1", "metadata": {"resourceVersion": "10596", ...}, ...}
+        }
+        ...
+        {
+          "type": "BOOKMARK",
+          "object": {"kind": "Pod", "apiVersion": "v1", "metadata": {"resourceVersion": "12746"}, }
```
Comment (liggitt) [inline content/en/docs/reference/using-api/api-concepts.md]: nit: no trailing comma

**Comment by liggitt** [comment]
lgtm
just a couple nits

File: content/en/docs/reference/using-api/api-concepts.md (hunk 3 above)
Comment (wojtek-t) [reply content/en/docs/reference/using-api/api-concepts.md]: done

File: content/en/docs/reference/using-api/api-concepts.md (hunk 2 above)
Comment (wojtek-t) [reply content/en/docs/reference/using-api/api-concepts.md]: done

**Comment by wojtek-t** [comment]
@liggitt - thanks a lot! Hopefully all comments should be applied.

**Pull request merged**